### PR TITLE
[SPARK-34701][SQL][FOLLOW-UP] Children/innerChildren should be mutually exclusive for AnalysisOnlyCommand

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This is a follow up to https://github.com/apache/spark/pull/32032#discussion_r620928086. Basically, `children`/`innerChildren` should be mutually exclusive for `AlterViewAsCommand` and `CreateViewCommand`, which extend `AnalysisOnlyCommand`. Otherwise, there could be an issue in the `EXPLAIN` command. Currently, this is not an issue, because these commands will be analyzed (children will always be empty) when the `EXPLAIN` command is run.

### Why are the changes needed?

To be future-proof where these commands are directly used.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Added new tsts

### CODE REVIEW

**Comment by imback82** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala [hunk 1]
```diff
@@ -77,7 +77,7 @@ case class CreateViewCommand(
     copy(plan = newChildren.head)
   }
 
-  override def innerChildren: Seq[QueryPlan[_]] = Seq(plan)
+  override def innerChildren: Seq[QueryPlan[_]] = if (isAnalyzed) Seq(plan) else Nil
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala:80]: shall we do it in `AnalysisOnlyCommand`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala [hunk 2]
```diff
@@ -46,5 +47,6 @@ trait AnalysisOnlyCommand extends Command {
   val isAnalyzed: Boolean
   def childrenToAnalyze: Seq[LogicalPlan]
   override final def children: Seq[LogicalPlan] = if (isAnalyzed) Nil else childrenToAnalyze
+  override def innerChildren: Seq[QueryPlan[_]] = if (isAnalyzed) childrenToAnalyze else Nil
```
Comment (imback82) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala:50]: @cloud-fan, updated. Do you also want to explicitly override this to be `Nil` for `CacheTable`, `CacheTableAsSelect` and `UncacheTable`?

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala:50]: Does it have real impact in EXPLAIN?

Comment (imback82) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala:50]: There is a change, but I think it's for better:
Before:
```
== Parsed Logical Plan ==
'CacheTableAsSelect tempTable, SELECT key FROM testData, false, false
+- 'Project ['key]
   +- 'UnresolvedRelation [testData], [], false

== Analyzed Logical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false, true

== Optimized Logical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false, true

== Physical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false
```

New:
```
== Parsed Logical Plan ==
'CacheTableAsSelect tempTable, SELECT key FROM testData, false, false
+- 'Project ['key]
   +- 'UnresolvedRelation [testData], [], false

== Analyzed Logical Plan ==
CacheTableAsSelect tempTable, SELECT key FROM testData, false, true
   +- Project [key#13]
      +- SubqueryAlias testdata
         +- View (`testData`, [key#13,value#14])
            +- SerializeFromObject [knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).key AS key#13, staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).value, true, false) AS value#14]
               +- ExternalRDD [obj#12]

== Optimized Logical Plan ==
CacheTableAsSelect tempTable, SELECT key FROM testData, false, true
   +- Project [key#13]
      +- SubqueryAlias testdata
         +- View (`testData`, [key#13,value#14])
            +- SerializeFromObject [knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).key AS key#13, staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).value, true, false) AS value#14]
               +- ExternalRDD [obj#12]

== Physical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false
```
WDYT?

Comment (imback82) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala:50]: And we can improve the EXPLAIN for physical plans as well as a future PR if needed.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala:50]: yea looks better!

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**